src/plugins/intel_cpu/src/utils/shape_inference/shape_inference.cpp

```diff
@@ -231,6 +231,7 @@ class entryFallback : public entryBase {
             for (size_t i = 0; i < local_op->get_input_size(); i++) {
                 if (dynamic_cast<ov::opset1::Parameter*>(local_op->get_input_node_ptr(i))) {
                     local_op->get_input_tensor(i).set_partial_shape(input_shapes[i].to_partial_shape());
```

Comment:
I propose to try next solution:
```suggestion
                if (auto parameter = dynamic_cast<ov::opset1::Parameter*>(local_op->get_input_node_ptr(i))) {
                    parameter.set_partial_shape(input_shapes[i].to_partial_shape());
```

Reply:
@ilyachur this suggestion doesn't work. This changes caused dynamic shape inference error, details can be found in simple PR [16230](https://github.com/openvinotoolkit/openvino/pull/16230)
I had reproduced the test error in local, get following output, 
![image](https://user-images.githubusercontent.com/52686861/225171496-469e8315-5c37-4102-8c87-430ecc446c62.png)
**Any other suggestions?**

Reply:
@jane-intel Could you take a look? Can we avoid the usage of this internal API?